rabbit-escape-engine/src/rabbitescape/engine/solution/SolutionRecorder.java

```diff
@@ -0,0 +1,51 @@
+package rabbitescape.engine.solution;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SolutionRecorder
+{
+    private List<CommandAction> commandInProgress;
+    private List<SolutionCommand> solutionInProgress;
+    
+    public SolutionRecorder()
+    {
+        commandInProgress = new ArrayList<CommandAction>();
+        solutionInProgress = new ArrayList<SolutionCommand>();
+    }
+
+    public void append( CommandAction a ) 
+    {
+        commandInProgress.add( a );
+    }
+    
+    public void appendStepEnd()
+    {
+        CommandAction[] aA = new CommandAction[commandInProgress.size()];
+        aA = commandInProgress.toArray( aA );
+        SolutionCommand newCmd = new SolutionCommand( aA );
+        int prevCmdIndex = solutionInProgress.size() - 1;
+        SolutionCommand prevCmd =   prevCmdIndex >= 0 
+                                  ? solutionInProgress.get( prevCmdIndex )
+                                  : null ;
+        SolutionCommand combCmd = SolutionCommand.tryToSimplify( prevCmd, newCmd );
+        if( null == combCmd)
+        {
+            solutionInProgress.add( newCmd );
+        }
+        else
+        { // Replace previous with combined
+            solutionInProgress.set( prevCmdIndex, combCmd );
+        }
+        
+        // Prepare to collect actions in the next step.
+        commandInProgress = new ArrayList<CommandAction>();
+    }
```

Comment:
Quite possibly. Some stuff should move from the text package to SolutionRecorder. I will look at that.